packages/trader/src/AppV2/Components/DraggableList/draggable-list.tsx

```diff
@@ -0,0 +1,161 @@
+    onDrag?: (categories: TDraggableListCategory[]) => void;
+};
+
+const DraggableList: React.FC<TDraggableListProps> = ({ categories, onRightIconClick, onAction, onDrag }) => {
+    const [category_list, setCategoryList] = React.useState(categories);
+    const [draggedItemId, setDraggedItemId] = React.useState<string | null>(null);
+
+    const handleOnDragEnd = (result: DropResult) => {
+        setDraggedItemId(null);
+        if (!result.destination) return;
+
+        const new_category_list = updateCategoriesWithDragResult(category_list, result);
+        setCategoryList(new_category_list);
+        onDrag?.(new_category_list);
+    };
+
+    const handleOnDragStart = (start: DragStart) => setDraggedItemId(start.draggableId);
+
+    React.useEffect(() => setCategoryList(categories), [categories]);
+
+    return (
+        <DragDropContext onDragEnd={handleOnDragEnd} onDragStart={handleOnDragStart}>
+            {category_list.map(category => (
+                <DraggableCategory
+                    key={category.id}
+                    category={category}
+                    draggedItemId={draggedItemId}
+                    onRightIconClick={onRightIconClick}
+                    onAction={onAction}
+                />
+            ))}
+        </DragDropContext>
+    );
+};
+
+const DraggableCategory: React.FC<{
+    category: TDraggableListCategory;
+    draggedItemId: string | null;
+    onRightIconClick: (item: TDraggableListItem) => void;
+    onAction?: () => void;
+}> = ({ category, draggedItemId, onRightIconClick, onAction }) => (
+    <div className='draggable-list-category'>
+        <DraggableCategoryHeader title={category.title} button_title={category.button_title} onAction={onAction} />
+        <Droppable droppableId={category.id}>
+            {provided => (
+                <div
+                    {...provided.droppableProps}
+                    ref={provided.innerRef}
+                    className='draggable-list-category__droppable-area'
+                >
+                    <DraggableCategoryItems
+                        items={category.items}
+                        draggedItemId={draggedItemId}
+                        onRightIconClick={onRightIconClick}
+                    />
+                    {provided.placeholder}
+                </div>
+            )}
+        </Droppable>
+    </div>
+);
+
+const DraggableCategoryHeader: React.FC<{
+    title?: string;
+    button_title?: string;
+    onAction?: () => void;
+}> = ({ title, button_title, onAction }) => (
+    <div className='draggable-list-category-header'>
+        <Text size='sm' bold className='draggable-list-category-header-title'>
+            {title}
+        </Text>
+        {onAction && (
+            <Text size='sm' bold underlined className='draggable-list-category-header-button' onClick={onAction}>
+                {button_title || <Localize i18n_default_text='Done' />}
+            </Text>
+        )}
+    </div>
+);
```

Comment:
There are 4 components in this file. These guys deserve to have their own files in the same folder.
*Note: if you simply separate them, you don't have to worry about writing tests for each of them if your current tests are already covering their code.

Reply:
@maryia-deriv these are very small components and they don't serve any purpose without each other, so I don't think we need to split them into separate files. Also, it used to be one component, but I had to change it because of some sonarclown complain. Do you think it makes sense to search for code in between 4 different separate files just to make a quick fix? I don't think so...

Reply:
Sorry @maryia-deriv, but I respectfully disagree, as I mentioned above, these components do not serve any purpose without each other, as well as the function, so there's no point to split them in different files. I will attach a random stackoverflow reply with high ratio to illustrate my point: 
<img width="647" alt="image" src="https://github.com/user-attachments/assets/44ee41bb-4fbb-4c40-b533-7ca184d4e3e4">